jax/experimental/pallas/ops/tpu/megablox/gmm.py

```diff
@@ -0,0 +1,794 @@
+# Copyright 2024 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Grouped matrix multiplication kernels for TPU written in Pallas."""
+
+import functools
+from typing import Any, Callable
+
+import jax
+from jax import lax
+from jax.experimental import pallas as pl
+from jax.experimental.pallas import tpu as pltpu
+from jax.experimental.pallas.ops.tpu.megablox import common
+import jax.numpy as jnp
+
+
+partial = functools.partial
+
+
+def _validate_args(
+    *,
+    lhs: jnp.ndarray,
+    rhs: jnp.ndarray,
+    group_sizes: jnp.ndarray,
+    expected_rhs_dims: int = 3,
+) -> tuple[jnp.ndarray, jnp.ndarray, jnp.dtype]:
+  """Validates the arguments for the gmm function."""
+  # Validate 'lhs'.
+  if lhs.ndim != 2:
+    raise ValueError(f"Expected 2-tensor for 'lhs' but got {lhs.ndim}-tensor.")
+  common.assert_is_supported_dtype(lhs.dtype)
+
+  # Validate 'rhs'.
+  if rhs.ndim != expected_rhs_dims:
+    raise ValueError(
+        "Expected {expected_rhs_dims}-tensor for 'rhs' but got"
+        " {rhs.ndim}-tensor."
```

Comment:
Nit: adding f-strings

```suggestion
        f"Expected {expected_rhs_dims}-tensor for 'rhs' but got"
        f" {rhs.ndim}-tensor."
```

Reply:
Thanks!